### ISSUE TITLE
Support build analyze and profile for static site generate.

### ISSUE THREAD

**Issue by Fnxxxxo** [nuxt/nuxt#8641]
### Is your feature request related to a problem? Please describe.

Currently, nuxt cannot set analyze or profile for static generating.

### Describe the solution you'd like

Apply analyze and profile to generate cmd too.

### Describe alternatives you've considered

A third party plugin is OK.

### Additional context

Noop.

**Comment by Fnxxxxo** [comment]
@Atinux Thanks for your reply. Currently, I have to run both `nuxt-ts build -a --profile` and `nuxt-ts generate` to complete a single build and analyzing which is costly. I need a way to run `nuxt-ts generate -a --profile`

**Comment by pi0** [comment]
> analyze since we don’t hash the files
That's true. Without having named chunks, analyze output is hard to follow. If per-build analyze is intended (like with a CI/CD), i would recommend using a nuxt module that integrates [webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) (or directly inside nuxt.config) instead of depending on `--analyze` flag.

**Comment by Fnxxxxo** [comment]
@pi0 Get it. Thanks.

**Comment by danielroe** [comment]
We are approaching the Nuxt 2 EOL date (June 30, 2024) - see [this article](https://nuxt.com/blog/nuxt2-eol) for more information. I'm closing this issue as it's marked as a _Nuxt 2 related enhancement_ and it's not a critical issue.

That doesn't mean it might not be relevant for Nuxt 3. If it is, please feel free to open a new issue (or just comment, and I can reopen it). 🙏

Thank you for your understanding and for your contribution to Nuxt! 🎉
